projects/distributed-tracing/src/shared/dashboard/interaction/detail-sheet/detail-sheet-interaction.module.ts

```diff
@@ -2,13 +2,15 @@ import { NgModule } from '@angular/core';
 import { DashboardCoreModule } from '@hypertrace/hyperdash-angular';
 import { DetailSheetInteractionContainerComponent } from './container/detail-sheet-interaction-container.component';
 import { DetailSheetInteractionHandlerModel } from './detail-sheet-interaction-handler.model';
+import { DetailSheetInteractionHandlerService } from './detail-sheet-interaction-handler.service';
 
 @NgModule({
   imports: [
     DashboardCoreModule.with({
       models: [DetailSheetInteractionHandlerModel]
     })
   ],
-  declarations: [DetailSheetInteractionContainerComponent]
+  declarations: [DetailSheetInteractionContainerComponent],
+  providers: [DetailSheetInteractionHandlerService]
```

Comment:
It won't make a difference here, but I was reading somewhere about avoiding declaring providers this way. Instead, if you declare it in the other direction (i.e.  `@Injectable({providedIn: DetailSheetInteractionModule})`) then it allows it to be tree shaken if not used. We know this one is used, but good habit to get in assuming it works.

Reply:
I don't think it will ever get tree shaken. We include the model in the common tracing dashboard interactions module which is also imported in the dashboard modules. Also, I would prefer it to be this way till we make a sweeping changes across all similar services

Reply:
OK - although in general we should separate the idea of "will it be tree shaken" from "can it be shaken". We want to support the latter whenever possible since the former may change.